### PR TITLE
feat:(apps and services) technical user creation in autosetup

### PR DESCRIPTION
**Pull request by qxz2mqe**
## Description

adrole assigned to technicaluser profile.
## Why

The check was added so that if technical user has role then allow to create the record elase return.

## Issue

CPLP-2829.

## Checklist

Please delete options that are not relevant.

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [ ] I have performed [IP checks](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-04#checking-libraries-using-the-eclipse-dash-license-tool) for added or updated 3rd party libraries
- [ ] I have created and linked IP issues or requested their creation by a committer
- [x] I have performed a self-review of my own code
- [ ] I have successfully tested my changes locally
- [ ] I have added tests that prove my changes work
- [x] I have checked that new and existing tests pass locally with my changes
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by qxz2mqe** [comment]
@Phil91  I have corrected the unit test

**Comment by Phil91** [comment]
@qxz2mqe please add a unit test to check whether the newly added logic works, no service account should be created if the service account profile has no role configured

**Comment by ntruchsess** [comment]
please do not use special characters in branch-names. Github is complaining: 'The head ref may contain hidden characters: "feature/CPLP-2829\u2014autosetup-ignore-techuser-without-profile"'

**Review by ntruchsess — DISMISSED** [review]

File: src/marketplace/Offers.Library/Service/OfferSetupService.cs [hunk 1]
```diff
@@ -141,7 +141,7 @@ public async Task<OfferAutoSetupResponseData> AutoSetupOfferAsync(OfferAutoSetup
             throw new UnexpectedConditionException($"There should only be one or none technical user profile configured for {subscriptionId}");
         }
 
-        if (serviceAccountCreationInfo == null)
+        if (serviceAccountCreationInfo == null || !serviceAccountCreationInfo!.UserRoleIds.Any())
```
[the remark this answers is not in the document]
Comment (Phil91) [reply src/marketplace/Offers.Library/Service/OfferSetupService.cs]: @qxz2mqe please don't just resolve comments if nothing was done...

fixed the cleanup

**Review by Phil91 — APPROVED** [review]

**Pull request merged**